modules/google-maps/src/google-maps-overlay.js

```diff
@@ -66,18 +73,56 @@ export default class GoogleMapsOverlay {
   }
 
   /* Private API */
+  _createOverlay(map) {
+    const {VECTOR, UNINITIALIZED} = google.maps.RenderingType;
+    const renderingType = map.getRenderingType();
+    if (renderingType === UNINITIALIZED) {
+      return;
+    }
+    const isVectorMap = renderingType === VECTOR;
+    const OverlayView = isVectorMap ? google.maps.WebglOverlayView : google.maps.OverlayView;
+    const overlay = new OverlayView();
+
+    // Lifecycle methods are different depending on map type
+    if (isVectorMap) {
+      overlay.onAdd = () => {};
+      overlay.onContextLost = this._onContextLost.bind(this);
+      overlay.onContextRestored = this._onContextRestored.bind(this);
+      overlay.onDraw = this._onDrawVector.bind(this);
+    } else {
+      overlay.onAdd = this._onAdd.bind(this);
+      overlay.draw = this._onDrawRaster.bind(this);
+    }
+    overlay.onRemove = this._onRemove.bind(this);
```

Comment:
Can we define these methods in ES6 syntax? E.g.

```js
const {VECTOR, UNINITIALIZED} = google.maps.RenderingType;

class GoogleMapsOverlay {
  _createOverlay(map) {
    this.renderingType = map.getRenderingType();
  }

  onAdd() {
    if (this.renderingType !== VECTOR) {
      this._deck = createDeckInstance(...);
    }
  }

  draw() {
    // old API - draw raster
  }

  onDraw() {
    // new API - draw vector
  }

  onRemove() {
  }
}
```

Reply:
Sorry, I don't follow here. Are the methods `_onAdd`, `_onContextRestored` etc not already using the correct syntax? We need to bind them to the `google.maps.WebglOverlayView` object as per Google's docs